### ISSUE TITLE
bug(mat-menu): Nested menus incorrectly placed when custom element above trigger 

### ISSUE THREAD

**Issue by NilsDietrich** [angular/components#28112]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

13.3.9

### Description

When using a <mat-menu>, which contains an element with an other size than 48px, following nested menus open on wrong positions:

![image](https://github.com/angular/components/assets/61544566/b382ffa5-abb4-44a7-8c14-202d666d5724)

If the nested menu is too small it looks completly unrelated to the trigger.
This worked correctly until v13.3.9:

![image](https://github.com/angular/components/assets/61544566/fb4af34e-b5ca-427f-ac43-e99ad99bac71)

### Reproduction

StackBlitz link: https://stackblitz.com/edit/1j8wme?file=src%2Fapp%2Fmenu-nested-example.html
Steps to reproduce:
Simply put a <div style="height: 70px"></div> before the trigger of a nested menu

### Expected Behavior

It should open the nested menu at the correct position like in v13.

### Actual Behavior

The nested menu opens in a wrong place.

### Environment

- Angular: v14.0.0 to latest
- CDK/Material: v14.0.0 to latest
- Browser(s): Chrome v119
- Operating System: WIndows 11 Pro

**Comment by NilsDietrich** [comment]
Seems like only mat-menu-items where counted to the offset. For now I'm using a workaround by adding invisible mat-menu-items with the same height `<div mat-menu-item style="height: 70px; display: none"></div>`

**Comment by wagnermaciel** [comment]
Hi, this seems to be entering into a gray area of what's supported by the mat-menu because it looks like you're feeding unexpected content into the mat-menu. If you want to do some heavy customization, I'd recommend using the cdk-menu. If it's just something small, I'd recommend either wrapping it in a `<button mat-menu-item` or adding the `mat-menu-item` directive to the div directly

CDK Menu:
https://material.angular.io/cdk/menu/overview

Example:
https://stackblitz.com/edit/1j8wme-d4pawu?file=src%2Fapp%2Fmenu-nested-example.html
